### PR TITLE
Check return values on APIs that can fail

### PR DESCRIPTION
**Pull request by lgirdwood**
Make sure check return values and act on them.

### CODE REVIEW

File: src/drivers/intel/baytrail/ssp.c [hunk 1]
```diff
@@ -619,16 +620,22 @@ static int ssp_probe(struct dai *dai)
...
 #else
 	/* register our IRQ handler */
-	interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
-			   dai);
+	ret = interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
+				 dai);
 #endif
+	if (ret < 0) {
+		trace_ssp_error("SSP failed to allocate IRQ");
+		rfree(ssp);
```
Comment (plbossart) [inline src/drivers/intel/baytrail/ssp.c]: we may want to have a test on the ssp = rzalloc upfront as well?
It beats me that such allocations are not tested for.

File: src/drivers/intel/cavs/dmic.c [hunk 2]
```diff
@@ -1489,8 +1485,18 @@ static int dmic_probe(struct dai *dai)
 	dmic->state = COMP_STATE_READY;
 
 	/* register our IRQ handler */
-	interrupt_register(dmic_irq(dai), IRQ_AUTO_UNMASK, dmic_irq_handler,
-			   dai);
+	ret = interrupt_register(dmic_irq(dai), IRQ_AUTO_UNMASK,
```
Comment (plbossart) [inline src/drivers/intel/cavs/dmic.c]: same above, why dmic = rzalloc isn't tested is weird

File: src/drivers/intel/cavs/ssp.c [hunk 3]
```diff
@@ -906,8 +904,17 @@ static int ssp_probe(struct dai *dai)
 	ssp->state[DAI_DIR_CAPTURE] = COMP_STATE_READY;
 
 	/* register our IRQ handler */
-	interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
-			   dai);
+	ret = interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
```
Comment (plbossart) [inline src/drivers/intel/cavs/ssp.c]: and here.

File: src/drivers/intel/cavs/dmic.c [hunk 4]
```diff
@@ -1465,17 +1465,13 @@ static void dmic_irq_handler(void *data)
...
 	struct dmic_pdata *dmic;
+	int ret;
 
 	trace_dmic("dmic_probe()");
 
 	if (dai_get_drvdata(dai))
 		return -EEXIST; /* already created */
 
-	/* Enable DMIC power */
-	pm_runtime_get_sync(DMIC_POW, dai->index);
```
Comment (singalsu) [inline src/drivers/intel/cavs/dmic.c]: Your patch should not remove microphones power line control code, right?

File: src/drivers/intel/cavs/dmic.c (hunk 2 above)
Comment (singalsu) [reply src/drivers/intel/cavs/dmic.c]: There's a check for dmic == NULL after rzalloc() but the part of code needed to be expanded  in Github view. Did you mean that?

File: src/drivers/intel/baytrail/ssp.c (hunk 1 above)
Comment (lgirdwood) [reply src/drivers/intel/baytrail/ssp.c]: @plbossart teh rzalloc in this case is using the system heap, which either returns a ptr or panics. It's meant for one time only allocs that never need to be freed.

**Pull request merged**